### DISCUSSION TITLE
Why does an ExperimentalWarning flag get logged as an error?

### DISCUSSION THREAD

**Discussion by john-zaprite** [nodejs/help#5012] [Q&A]
Since I upgraded my Vercel project to Node v18x, I'm getting a lot of error logs in my console regarding the Fetch API:
`Error: (node:15) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time`

I understand this is happening since the 17.5 upgrade.

But why don't they get logged as Warnings instead? They are clearly not Errors.

**Comment by cjihrig** [comment]
I have not used Vercel, but do they treat everything sent to stderr as errors?

**Reply by john-zaprite** [reply]
I'm not sure. I posted over at Vercel too but only got a generic response.

But a search found this:
https://github.com/vercel/next.js/discussions/41124

The Vercel employee says they are using `console.warn`, but in NodeJs that's just an alias for `console.error`, which would explain to error log.

Do you know if this is true?

**Reply by cjihrig** [reply]
Yes, that is true.

**Reply by john-zaprite** [reply]
Right. Really curious about why that is? Seems easy to detect warnings and show them as such. Showing them as Errors seems deliberate so I'm curious about why that might be?

**Reply by cjihrig** [reply]
> Really curious about why that is?

There are a handful of `console` methods - `info()`, `log()`, `debug()`, `warn()`, `error()`, etc. Node.js can only map all of those to `stdout` and `stderr`.

> Showing them as Errors seems deliberate so I'm curious about why that might be?

This is something that Vercel needs to handle.

**Reply by john-zaprite** [reply]
Right thanks. So would Vercel have the ability to map a warning from `stderr` back to a `console.warn` in their logs? Or can they not tell the difference?

I don't fully understand all this, but seems like I need to get back in touch with Vercel. 

Thanks!

**Reply by cjihrig** [reply]
Based on the contents of https://github.com/vercel/next.js/discussions/41124 that you linked to earlier, it seems like they should be able to.

EDIT: Actually, that thread seems to be about warnings that they generate themselves, not Node.js. Either way, there are ways to do it.
